ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/LimitPayloadSize.java

```diff
@@ -25,15 +32,21 @@ public static LimitPayloadSize toMB(int mb) {
   }
 
   @Override
-  protected PubsubMessage processElement(PubsubMessage message) throws Exception {
-    message = PubsubConstraints.ensureNonNull(message);
-    int numBytes = message.getPayload().length;
-    if (numBytes > maxBytes) {
-      countPayloadTooLarge.inc();
-      throw new PayloadTooLargeException("Message payload is " + numBytes + "bytes, larger than the"
-          + " configured limit of " + maxBytes);
-    }
-    return message;
+  public Result<PCollection<PubsubMessage>, PubsubMessage> expand(
+      PCollection<PubsubMessage> elements) {
+    return elements.apply(
+        MapElements.into(TypeDescriptor.of(PubsubMessage.class)).via((PubsubMessage message) -> {
+          message = PubsubConstraints.ensureNonNull(message);
+          int numBytes = message.getPayload().length;
+          if (numBytes > maxBytes) {
+            countPayloadTooLarge.inc();
+            throw new RuntimeException(new PayloadTooLargeException("Message payload is " + numBytes
+                + "bytes, larger than the" + " configured limit of " + maxBytes));
+          }
+          return message;
+        }).exceptionsInto(TypeDescriptor.of(PubsubMessage.class))
+            .exceptionsVia((ExceptionElement<PubsubMessage> ee) -> FailureMessage.of(this,
+                ee.element(), ee.exception())));
```

Comment:
We could reraise `ee.exception` here so that we can catch only the expected case (`PayloadTooLarge`), which is a potential benefit we get from this pattern vs. what we were doing in MapElementsWithFailures.

Also, we could sidestep the wrapped exception problem here by inspecting `ee.exception().getCause()`.